### ISSUE TITLE
[DISCUSSION] Lawnicons not applying dark-theme icon pack to other launchers

### ISSUE THREAD

**Issue by lettersforword365** [LawnchairLauncher/lawnicons#1154]
### About this discussion

I've just updated to the latest version (2.0.0) on my Android 11 (OxygenOS 11) device.

The update means icons can be applied to launchers beside Lawnchair.

I read that:
> [Lawnicons will have a dark or white background depending on the system theme](https://github.com/LawnchairLauncher/lawnicons/releases/tag/v2.0.0)

My device has a systemwide dark theme which supports all apps with this feature. However, I am unable to view dark icons regardless of the system theme's state (both dark and light).

Applying Lawnicons to launchers (except Lawnchair) shows light-themed icons:

![](https://user-images.githubusercontent.com/98777122/231280064-f13e3532-1845-4ebe-9256-09b569e64d37.jpg)
*Lawnicons applied to other launcher (systemwide dark theme)*

I also noticed that the `About` tab of the Lawnicons app shows the app icon in a light background when the systemwide dark theme is enabled and vice versa:

![](https://user-images.githubusercontent.com/98777122/231280168-74028ad4-e675-4b78-bcf4-1034ad760ef7.jpg)
*Lawnicons About tab (systemwide dark theme)*

![](https://user-images.githubusercontent.com/98777122/231280246-36e5be5c-d825-4cd7-8646-8628d7cf8561.jpg)
*Lawnicons About tab (systemwide light theme)*

Are the icons (when applied to other launchers) also expected to be black and white for devices that don't natively support material you (Android 11 and below), or does the app have support for wallpaper-based colouring on all devices (like Lawnchair)? I am unable to produce dark-themed icons, so I am unable to tell myself.

**Comment by dhirenvasnani** [comment]
I am also experiencing same issue, MIUI 12.5 Android 11.

**Comment by ereinstein** [comment]
Pixel 7 Pro with GrapheneOS and Lawnchair:

When I set an application's icon manually, it does not adapt to changes between light and dark themes, e.g. when I set the light themed icon while my system theme is light, it remains light even if my system switches to a dark theme afterwards and vice versa. 

When I press and hold on the icon in order to change it via "Customize", the icon shown in the pop-up window respects the system theme regardless of the icon theme on the home screen. 

Everything works perfectly for icons that are set automatically because the app itself has an icon in the icon pack.

**Comment by QKIvan** [comment]
The same in Android13,OnePlus 8 Pro,Nameless Launcher

**Comment by SuperDragonXD** [comment]
The core issue in the first post is fixed in the latest version (2.1). The colors for pre-A12 devices was reversed, and now fixed in that version.

As for the themed icons not properly changing colors (A12+ only), try restarting Lawnchair, change the icon pack (to trigger icon cache reset), re-apply the icons.

**Comment by E-D-W-I-N** [comment]
> As for the themed icons not properly changing colors (A12+ only), try restarting Lawnchair, change the icon pack (to trigger icon cache reset), re-apply the icons. 

Yeah, It works if you change the icon pack a few times and restart Lawnchair, but still it's really bad UX. I hope there's a way to clear cache automatically on wallpaper or system color change

**Comment by SuperDragonXD** [comment]
Since there is no further activity, I'm closing this issue. If the issue still persists on the latest nightly version, feel free to create a new one with additional details.
